### PR TITLE
feat(share_plus): Use XFile.name whenever possible, shorten UUID filenames, and improve I/O throughput

### PR DESCRIPTION
**Pull request by LBYPatrick**
## Description

- Uses ``XFile.name`` whenever possible, instead of always generating a UUID filename.
- Patches the MIME extension issue originated from the mime package (should remove this in the future, noted with a TODO in code)
- Improves I/O throughput of ``MethodChannelShare._getFile`` with parallel FS writes.
- Shortens the UUID filenames without sacrificing, if not improving, the ability to prevent filename collision.

## Related Issues

- **Related #1548*

## Checklist

- [x] I read the [Contributor Guide](https://github.com/fluttercommunity/plus_plugins/blob/main/CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
- [x] I titled the PR using [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0).
- [x] I did not modify the `CHANGELOG.md` nor the plugin version in `pubspec.yaml` files.
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate that with a `!` in the title as explained in [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0)).
- [x] No, this is *not* a breaking change.

### CODE REVIEW

**Comment by LBYPatrick** [comment]
Another issue worth noting is that per [IANA](https://www.iana.org/assignments/media-types/media-types.xhtml#image), ``image/jpe`` is not longer a valid MIME type, which might explain why most software do not handle ``image/jpe`` and MDN (for JavaScript) never mentions ``image/jpe``.

**Comment by LBYPatrick** [comment]
android_integration-test is currently failing just like other PRs. May I skip integration tests for API level 22 or should I wait for #2587 to merge first (I believe it resolves the issue?)

File: packages/share_plus/share_plus_platform_interface/lib/method_channel/method_channel_share.dart [hunk 1]
```diff
@@ -177,42 +177,52 @@ class MethodChannelShare extends SharePlatform {
...
+  Future<XFile> _getFile(XFile file, {String? tempRoot}) async {
+    if (file.path.isNotEmpty) {
+      return file;
     } else {
-      return files;
+      tempRoot ??= (await getTemporaryDirectory()).path;
+      var extension = extensionFromMime(file.mimeType ?? 'octet-stream');
+
+      // TODO: As soon as the mime package fixes the image/jpe issue, remove this line immediately
+      extension = extension == "jpe" ? "jpeg" : extension;
```
Comment (miquelbeltran) [inline packages/share_plus/share_plus_platform_interface/lib/method_channel/method_channel_share.dart:194]:
```suggestion
      // TODO: As soon as the mime package fixes the image/jpe issue, remove this line immediately
      // Reference: https://github.com/dart-lang/mime/issues/55
      extension = extension == "jpe" ? "jpeg" : extension;
```

File: packages/share_plus/share_plus_platform_interface/lib/method_channel/method_channel_share.dart [hunk 2]
```diff
@@ -177,42 +177,52 @@ class MethodChannelShare extends SharePlatform {
...
+
+      //By having a UUID v4 folder wrapping the file
+      //This path generation algorithm will not only minimize the risk of name collision but also ensure that the filename
+      //is not ridiculously long such that some platforms might not show the extension but ellipses
+      //which the user needs
+      //
+      //More importantly it allows us to use real filenames when available
+      final tempSubfolderPath = "$tempRoot/${const Uuid().v4()}";
+      await Directory(tempSubfolderPath).create(recursive: true);
```
Comment (miquelbeltran) [inline packages/share_plus/share_plus_platform_interface/lib/method_channel/method_channel_share.dart:203]: Good idea!

**Review by miquelbeltran — APPROVED** [review]
Thanks for the contribution!

See if you can do the suggestion change and add the reference url in the comment, so we can easily find the mime package ticket.

**Comment by miquelbeltran** [comment]
> android_integration-test is currently failing just like other PRs. May I skip integration tests for API level 22 or should I wait for #2587 to merge first (I believe it resolves the issue?)

It's a very flaky test, no worries about it, only the jobs marked as "required" are necessary

**Comment by miquelbeltran** [comment]
> See if you can do the suggestion change and add the reference url in the comment, so we can easily find the mime package ticket.

Already did that, hopefully you don't mind!

**Pull request merged**